crates/bevy_diagnostic/src/system_information_diagnostics_plugin.rs

```diff
@@ -14,12 +14,63 @@ use bevy_app::prelude::*;
 /// # See also
 ///
 /// [`LogDiagnosticsPlugin`](crate::LogDiagnosticsPlugin) to output diagnostics to the console.
-#[derive(Default)]
-pub struct SystemInformationDiagnosticsPlugin;
+pub struct SystemInformationDiagnosticsPlugin {
+    pub max_history_length: usize,
+}
+
+impl Default for SystemInformationDiagnosticsPlugin {
+    fn default() -> Self {
+        Self {
+            max_history_length: 20,
+        }
+    }
+}
+
 impl Plugin for SystemInformationDiagnosticsPlugin {
     fn build(&self, app: &mut App) {
-        app.add_systems(Startup, internal::setup_system)
-            .add_systems(Update, internal::diagnostic_system);
+        #[cfg(all(
+            any(
+                target_os = "linux",
+                target_os = "windows",
+                target_os = "android",
+                target_os = "macos"
+            ),
+            not(feature = "dynamic_linking")
+        ))]
+        {
+            use crate::RegisterDiagnostic;
+
+            app.register_diagnostic(
```

Comment:
It looks like this is fixing a bug where `SystemInformationDiagnosticsPlugin` doesn't work if it's the only diagnostics plugin in the app?

Is that right?

It seems unfortunate that the cfg directives are now repeated here. Would it be possible to re-implement `internal::setup_system` as a normal function that takes `&mut App` or something?

Reply:
I'll check it out when I get back home

Reply:
yep it also fix the bug with `SystemInformationDiagnosticsPlugin`, I updated my code to avoid multiples `cfg`